app.rb

```diff
@@ -0,0 +1,99 @@
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+require_relative 'person'
+
+class App
+  attr_accessor :books, :persons, :rentals
+
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  # list all books
+  def list_books
+    puts 'List of books'
+    @books.each { |book| puts "title : #{book.title} by author : #{book.author}" }
+  end
+
+  def list_persons
+    @persons.each do |person|
+      if person.instance_of?(Teacher)
+        puts "[Teacher]Age: #{person.age} Name: #{person.name}, ID: #{person.id}"
+      else
+        puts "[Student]Age: #{person.age},Name: #{person.name}, ID: #{person.id} "
+      end
+    end
+  end
+
+  # create person
+  def create_person
+    print 'Do you want to add a student (1) or a teacher (2)? [Insert the number]: '
+    is_student = gets.chomp.strip.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    person_name = gets.chomp
+    case is_student
+
+    when 1
+      puts 'Has Parent Permission [Y/N]'
+      permission = gets.chomp.downcase == 'y'
+      create_student(age, person_name, permission)
+    when 2
+      create_teacher(age, person_name)
+    end
+    puts 'Person added successfully'
+  end
+
+  def create_student(age, person, permission)
+    persons << Student.new(age, person, parent_permission: permission)
+  end
+
+  def create_teacher(age, person)
+    puts 'Specialization'
+    specialization = gets.chomp
+    @persons << Teacher.new(age, specialization, person)
+  end
+
```

Comment:
- [ ] Good job so far 👏 However, I am currently unable to create a teacher. I noticed that this functionality was working as expected during the last review. Since this issue was not present in the last review, please make sure to fix it.

During last review | Current implementation
---| ---|
![image](https://github.com/supreeti/OPPS-Library-Management-System/assets/106119105/f5e6b651-9921-464c-8147-8053ab3f76de) | ![image](https://github.com/supreeti/OPPS-Library-Management-System/assets/106119105/60ba49f4-4dbd-4021-984a-80c7bfb58dfc)

Reply:
Fixed.